pkg/datagatherer/k8s/dynamic_test.go

```diff
@@ -1009,3 +1086,200 @@ func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
 		return true
 	}
 }
+
+func TestRemoveUnstructuredKeys(t *testing.T) {
+	t.Run("remove single key", run_TestRemoveUnstructuredKeys(tc_RemoveUnstructuredKeys{
+		givenPath:    []string{"metadata", "annotations"},
+		givenExclude: []string{"^key1$"},
+		givenObj: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name": "foo",
+				"annotations": map[string]interface{}{
+					"key1": "value1",
+					"key2": "value2",
+				},
+			},
+		},
+		expectObj: map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"name": "foo",
+			},
```

Comment:
Shouldn't `key2` remain ?
```suggestion
			"metadata": map[string]interface{}{
				"name": "foo",
				"annotations": map[string]interface{}{
					"key2": "value2",
				},
			},
```

Reply:
Thanks for noticing this; I had forgotten to write the assertion, which led to this 🤦  I fixed this and renamed `key1` to `toexclude` to make it clear which fields are meant to disappear in each test.